core/services/workflows/metering/metering.go

```diff
@@ -580,6 +591,26 @@ func (r *Report) FormatReport() *protoEvents.MeteringReport {
 	return protoReport
 }
 
+// isRetryableError determines if an error is retryable based on gRPC status codes
+func isRetryableError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	st, ok := status.FromError(err)
+	if !ok {
+		// If we can't determine the status, don't retry
+		return false
+	}
+
+	switch st.Code() {
```

Comment:
[status.Code](https://pkg.go.dev/google.golang.org/grpc/status#Code) returns `codes.OK` for `nil` and `codes.Unknown` for unrecognized, so this could be simplified:
```suggestion
	switch status.Code(err) {
```

Reply:
thanks, yes: 225a954585c54fd205faf73b93ba1a856a676dbd